### PR TITLE
Rotating tips and homepage layout

### LINKED ISSUE #83: Create dashboard page and Rotating Tips component
https://linear.app/trap-garden/issue/TG-57/create-dashboard-page-and-rotating-tips-component

### PR DESCRIPTION
**Pull request by SashankBalusu**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

I hope you all are ready for this sprint!!!! Boy was it a ride! There's two main things in this sprint
1. the rotating tip logic and component
- with the help of cht gpt and cs 70 we now have an algorithm that shuffles through the existing tips to ensure that a. we get a new tip every day until there are no tips remaining and we can restart and b. the cycling through the tips isnt in the same order each time
2. general placeholders for the rest of the future homepage content
### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

Screenshot 2025-03-15 at 12 30 54 PM
BANGGGGG golly gee look at that front end!!!

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @kylezryr

### CODE REVIEW

File: app/add-details/page.tsx
Comment (kylezryr) [inline app/add-details/page.tsx:1]: remove this file from pr

File: app/add-details/styles.ts
Comment (kylezryr) [inline app/add-details/styles.ts:1]: remove this file

File: components/ConfirmationModal/index.tsx
Comment (kylezryr) [inline components/ConfirmationModal/index.tsx:1]: remove this file

File: components/ConfirmationModal/styles.ts
Comment (kylezryr) [inline components/ConfirmationModal/styles.ts:1]: remove this file

File: components/NavColumn/index.tsx
Comment (kylezryr) [inline components/NavColumn/index.tsx:1]: remove this file

File: components/PlantCard/styles.ts
Comment (kylezryr) [inline components/PlantCard/styles.ts:1]: remove this file

File: components/PlantDetails/index.tsx
Comment (kylezryr) [inline components/PlantDetails/index.tsx:1]: remove this file

File: components/PlantDetails/styles.ts
Comment (kylezryr) [inline components/PlantDetails/styles.ts:1]: remove this file

File: app/dashboard/styles.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  margin-bottom: 4px;
+`;
+
+export const Title = styled.h1`
+  color: ${COLORS.shrub};
+  font-family: Lexend;
+  font-size: 16px;
```
Comment (kylezryr) [inline app/dashboard/styles.ts]: use rem for font-size, so 16px = 1rem

File: app/dashboard/styles.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+export const Header = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  margin-bottom: 4px;
+`;
+
+export const Title = styled.h1`
+  color: ${COLORS.shrub};
+  font-family: Lexend;
```
Comment (kylezryr) [inline app/dashboard/styles.ts]: you don't need to assign the font-family prop to any of the styled components, it is already applied by default

File: app/dashboard/styles.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+`;
+
+export const Header = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  margin-bottom: 4px;
+`;
+
+export const Title = styled.h1`
```
Comment (kylezryr) [inline app/dashboard/styles.ts]: instead of making these styled components of the default `h1` / `p` etc. tags, you can use the `H1`, `H2`, `P1` components inside of `styles/text.ts` instead and apply stylings to them, like this:

```
//import H2 from styles/text
export const Title = styled(H2)`
....
```

this way, you don't need the font-family, font-style, and line-height props. you only maybe need to adjust fontweight and color.

File: app/dashboard/styles.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+export const InfoContainer = styled.div`
+  display: flex;
+  align-items: center;
+  margin-bottom: 12px;
+`;
+
+export const InfoHead = styled.p`
+  color: ${COLORS.midgray};
+  font-family: Lexend;
+  font-size: 10px;
```
Comment (kylezryr) [inline app/dashboard/styles.ts:50]: rem for font-size

File: app/dashboard/styles.ts [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+export const Circle = styled.div`
+  color: ${COLORS.midgray};
+  width: 10px;
+  height: 10px;
+  border: 1px solid #666;
+  border-radius: 50%;
+  margin-right: 10px;
+`;
+
+export const Greeting = styled.h2`
```
Comment (kylezryr) [inline app/dashboard/styles.ts]: H2 from styles/text

File: app/dashboard/styles.ts [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+  font-family: Lexend;
+  font-size: 28px;
+  font-style: normal;
+  font-weight: 400;
+  line-height: normal;
+  margin-bottom: 16px;
+`;
+
+export const DashboardTitle = styled.p`
```
Comment (kylezryr) [inline app/dashboard/styles.ts]: P2 from styles/text

File: components/SingleTip/index.tsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+// Define the category type with specific values
+
+// Map category values to headers
+const categoryHeaders: Record<Category, string> = {
+  'Helpful Flowers for Your Garden': 'Helpful Flowers',
+  'Water Management': 'Water Tips',
+  Mulching: 'Mulching Tips',
+  Harvesting: 'Harvesting Tips',
+  Planting: 'Planting Tips',
+  Weeding: 'Weeding Tips',
```
Comment (kylezryr) [inline components/SingleTip/index.tsx]: shouldn't these props (`Mulching, Harvesting, Planting, Weeding`) also be strings? why are only the first two strings and the others not strings?

File: components/SingleTip/styles.ts [hunk 8]
```diff
@@ -0,0 +1,31 @@
+import styled from 'styled-components';
+import COLORS from '@/styles/colors';
+
+export const Card = styled.div`
+  background-color: white;
+  box-shadow: 2px 0px 8px 0px rgba(0, 0, 0, 0.1);
+  border-radius: 10px;
+  padding: 24px;
```
Comment (kylezryr) [inline components/SingleTip/styles.ts]: the Figma styling slightly changed so the padding should be `20px 20px 16px 16px`

File: components/SingleTip/styles.ts [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+  height: 109px;
+`;
+
+export const Title = styled.h2`
```
Comment (kylezryr) [inline components/SingleTip/styles.ts]: use `P1` from styles/text

File: components/SingleTip/styles.ts [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+  font-family: Lexend;
+  font-size: 16px;
+  font-style: normal;
+  font-weight: 300;
+  line-height: normal;
+  margin-bottom: 6px;
+  text-align: center;
+`;
+
+export const Text = styled.p`
```
Comment (kylezryr) [inline components/SingleTip/styles.ts:37]: check the updated figma designs below, we need to indent the body text slightly now: 

image

File: types/schema.d.ts [hunk 11]
```diff
@@ -58,3 +58,32 @@ export interface OwnedPlant {
...
+  initialValue?: string;
+}
+
+export interface PlantTip {
+  id: UUID;
+  category: Category;
+  body_text: string;
+}
+
+export type Category =
```
Comment (kylezryr) [inline types/schema.d.ts]: could we rename this to `TipCategory` just to avoid confusion with other types if it comes up in the future?

**Review by kylezryr — CHANGES_REQUESTED** [review]
looks good overall, just a couple of styling changes to make. functionality is working all good so far, nice one

**Review by kaleenchen — APPROVED** [review]

**Review by kylezryr — APPROVED** [review]
looks good working well thanks g
